frontend/src/components/dashboard/aliases/CustomAddressGenerationModal.tsx

```diff
@@ -0,0 +1,467 @@
+
+  return (
+    <span className={styles.wrapper}>
+      <span className={styles.tooltip}>
+        <div className={styles.errors}>
+          <ErrorStateIcons address={address} errorState={containsUppercase} />
+          <p>
+            {l10n.getString(
+              "error-alias-picker-prefix-invalid-uppercase-letters",
+            )}
+          </p>
+        </div>
+        <div className={styles.errors}>
+          <ErrorStateIcons address={address} errorState={containsSymbols} />
+          <p>{l10n.getString("error-alias-picker-prefix-invalid-symbols")}</p>
+        </div>
+      </span>
+    </span>
+  );
+};
+
+type ErrorStateProps = {
+  errorState: boolean;
+  address: string;
+};
+
+const ErrorStateIcons = (props: ErrorStateProps) => {
+  const { errorState, address } = props;
+  const l10n = useL10n();
+
+  return (
+    <>
+      <div className={styles["error-icons"]}>
+        {address === "" ? (
+          <BulletPointIcon alt="" className={styles["bullet-icon"]} />
+        ) : !errorState ? (
+          <CheckIcon
+            alt={l10n.getString("error-state-valid-alt")}
+            className={styles["check-icon"]}
+          />
+        ) : (
+          <InvalidIcon
+            alt={l10n.getString("error-state-invalid-alt")}
+            className={styles["close-icon"]}
+          />
+        )}
+      </div>
+    </>
+  );
+};
+
+type InvalidPrefixProps = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const InvalidPrefixBanner = (props: InvalidPrefixProps) => {
+  const l10n = useL10n();
+
+  return (
+    <div
+      className={`${styles["invalid-address-wrapper"]} ${
+        props.isOpen ? styles["active"] : null
+      }`}
+    >
+      <div className={styles["invalid-address-msg"]}>
+        <div className={styles["left-content"]}>
+          <ErrorTriangleIcon alt="" className={styles["prefix-error-icon"]} />
+          <p>{l10n.getString("error-alias-picker-prefix-invalid")}</p>
+        </div>
+        <button
+          onClick={props.onClose}
+          className={`${styles["prefix-error-icon"]}  ${styles["close-button"]}`}
+        >
+          <CloseIconNormal alt={l10n.getString("close-button-label-alt")} />
+        </button>
+      </div>
+    </div>
+  );
+};
+
+export function isAddressValid(address: string): boolean {
+  // Regular expression:
+  //
+  //   ^[a-z0-9]  Starts with a lowercase letter or number;
+  //
+  //   (...)?     followed by zero or one of:
+  //
+  //              [a-z0-9-.]{0,61} zero up to 61 lowercase letters, numbers, hyphens, or periods, and
+  //              [a-z0-9]         a lowercase letter or number (but not a hyphen),
+  //
+  //   $          and nothing following that.
+  //
+  // All that combines to 1-63 lowercase characters, numbers, or hyphens,
+  // but not starting or ending with a hyphen, aligned with the backend's
+  // validation (`valid_address_pattern` in emails/models.py).
+  return /^[a-z0-9]([a-z0-9-.]{0,61}[a-z0-9])?$/.test(address);
+}
```

Comment:
Question/Praise (non-blocking): Awesome job with this detailed comment. Also, wonder if there is a better place for this function.